pkg/kubelet/server/stats/volume_stat_calculator.go

```diff
@@ -91,10 +91,24 @@ func (s *volumeStatCalculator) GetLatest() (PodVolumeStats, bool) {
 func (s *volumeStatCalculator) calcAndStoreStats() {
 	// Find all Volumes for the Pod
 	volumes, found := s.statsProvider.ListVolumesForPod(s.pod.UID)
-	if !found {
+	blockVolumes, bvFound := s.statsProvider.ListBlockVolumesForPod(s.pod.UID)
+	if !found && !bvFound {
```

Comment:
Is there a way to have volumes (and no block volumes) or vice versa? It seems if that was the case then no metrics would be calculated based on this condition.

Reply:
Yes, you can have none, either, or both filesystem-volumes and block-volumes. In case there are `no filesystem-volumes` **and** `no block-volumes`, stats do not need to get gathered. In case some filesystem-volumes or block-volumes exist, stats will get gathered.

Reply:
I think I misread this. This makes sense, thank you for clarifying.